force-app/main/default/lwc/lookUp/lookUp.js

```diff
@@ -0,0 +1,98 @@
+import { LightningElement, wire, api, track } from 'lwc';
+import getRecordList from '@salesforce/apex/LookUpController.getRecordList';
+
+export default class Lookups extends LightningElement {
+
+    @api
+    sObjectApiName;
+
+    @api
+    addInfoFieldApiName = '';
+
+    @track
+    searchKey = '';
+
+    @track
+    selectedRecord;
+
+    @track
+    records;
+
+    @track 
+    error;
+
+    @track 
+    flag = false;
+
+    @wire(getRecordList, {
+        sObjectName : '$sObjectApiName', 
+        addInfo : '$addInfoFieldApiName',
+        searchKey : '$searchKey'
+    })
+    getRecords({
+        data, error
+    }) {
+        if(data) {
+            if(data.length > 0) {
+                this.records = data.map(element => {
+                    const {Id, Name } = element;
+                    return {Id, Name, AddInfoData : element[this.addInfoFieldApiName]};
+                });
+            } else {
+                this.records = null;
+            }
+        } else if (error) {
+            this.error = error;
+        }
+    }
+
+    get boxClass() {
+        return 'slds-combobox slds-m-left_small slds-m-right_small slds-dropdown-trigger slds-dropdown-trigger_click ' +
+            (this.flag ? 'slds-is-open' : '');
+    }
+
+    get changeMarginOfRecord() {
+        const margin = 'slds-m-top_x-small';
+        return this.addInfoFieldApiName ? `slds-listbox__option-text slds-m-left_xxx-small slds-listbox__option-text_entity` :
+            `slds-listbox__option-text slds-m-left_xxx-small slds-listbox__option-text_entity ${margin}`;
+    }
+
+    get changeIconLocate() {
+        return this.selectedRecord ? 'slds-combobox__form-element slds-input-has-icon slds-input-has-icon_left-right' :
+            'slds-combobox__form-element slds-input-has-icon slds-input-has-icon_right'
+    }
+
+    get selectValue() {
```

Comment:
@TsSlava sounds like method which selects something, but actually is property getter